src/lib/query-store.ts

```diff
@@ -32,23 +32,27 @@ export function createQueryStore(prop: any, initialValue: any) {
 }
 
 export function pageQueryStore<T extends object, Key extends keyof T>(props: T) {
-   var query = undefined;
+   let query = undefined;
+
+   const intParams = ['page'];
+
+   const processProp = (name, value) => intParams.includes(name) ? parseInt(value, 10) : value;
+
    return {
       subscribe: (h): T => {
          return page.subscribe((p) => {
             query = queryString.parse(p.query.toString());
-            h(Object.keys(props).map((p) => {
-               return {
-                  [p]: query[p] || props[p]
-               }
-            }).reduce((a, b) => { return { ...a, ...b } }));
+
+            h(Object.keys(props).map((p) => ({
+                  [p]: processProp(p, query[p] || props[p])
+            })).reduce((a, b) => ({ ...a, ...b }), {}));
```

Comment:
Kinda unrelated but this is just straight up unreadable, even as somebody who is proficient in JS. This should at least have a comment outlining the input and result.

Reply:
I agree, some comments on how you solved this issue are most welcome to keep the whole code-base maintenable @motzel 😊

Reply:
If I got it right and ES2019 is the Target, fromEntries would probably be the better-readable alternative:

`h(Object.fromEntries(Object.keys(props).map(prop => [prop, query[p] || props[p]]));`

Or even better yet, just building it with a for loop.

Reply:
Well, I didn't change the original map/reduce solution.  The only thing I added was parseInt() for "page" property.

![image](https://user-images.githubusercontent.com/1758707/142941221-33e6885f-9a3c-4c0b-96c1-2b81051d086c.png)

And then I use processProp() in the few places where there was a direct prop  assignment.

Reply:
Yeah I'm aware you didnt change it, which is why I said its kinda unrelated - Could still be improved.

Reply:
Do you think that using Object.fromEntries() as you suggested will be a good solution? Personally, I have no problem with the readability of either 😁 The question is what is the browser target? Because Object.fromEntries() is available in Chrome since version 79 and Firefox since 63. Object.keys() works even on IE xD

Reply:
Hence why I said probably best to just build it with a for loop. Would end up being 1-2 lines more than the (formatted) reduce solution and significantly more readable.

```
const outObj = {};
for(const prop in props) {
	outObjs[prop] = query[prop] || props[prop];
}
h(outObj);
```

Reply:
TBH I think this store needs a whole rethink and rewrite. It's used in different places in the codebase and the structure looks different every time. This was supposed to be a quick fix for a problem that affects all users on mobile.

Reply:
The best solution to this problem would probably be to create an interface for each use case and use strong types. But I'm definitely not the right person for this task, because I haven't dealt with Typescript at all.

Reply:
I'm looking into a better solution atm, that is a bit more than a bandaid fix. It already has some type safety but it could probably be improved. If there are any more issues you see with it I'll look into them as well. Also as a general note, I'll try and make it more readable and chuck some comments in there so it doesn't look like black magic at times.

Reply:
#35 Should solve the issue and hopefully improve readability/type safety

Reply:
So this PR can now be closed

Reply:
Yep, if you're happy that the other PR solves the issues/concerns you had

Reply:
Sure!